cli/edp/cli.py

```diff
@@ -53,58 +54,55 @@ def formatPos(_self):
 def cli():
     pass
 
-@cli.command('ingest', help='Ingest data')
-@click.option('-p', '--project', default=None, help='the project id', required=True)
-@click.option('-c', '--cycle', default=None, help='the cycle id', required=True)
-@click.option('-d', '--dir', help='path to batch to ingest',
-              type=click.Path(exists=True, dir_okay=True, file_okay=False, readable=True), default='.')
-@click.option('-u', '--api-url', default='http://localhost:8080/api/v1', help='RESTful API URL '
-                   '(e.g https://girder.example.com/api/v1)')
-@click.option('-k', '--api-key', envvar='GIRDER_API_KEY', default=None,
-              help='[default: GIRDER_API_KEY env. variable]', required=True)
-def _ingest(project, cycle, api_url, api_key, dir):
-    gc = GC(api_url=api_url, api_key=api_key)
-
-    # Try to get edp data folder
-    data_folder = gc.resourceLookup('/collection/edp/data')
-
-    # Create a private folder
-    if data_folder is None:
-
-        me = gc.get('/user/me')
-        private_folder = next(gc.listFolder(me['_id'], 'user', 'Private'))
+def _ingest_batch(gc, data_folder, project, cycle, dir, public, summary_func):
 
-        data_folder = gc.listFolder(private_folder['_id'], 'folder', name='edp')
+    # Loady summary function
+    if summary_func is not None:
         try:
-            data_folder = next(data_folder)
-        except StopIteration:
-            data_folder = gc.createFolder(private_folder['_id'], 'edp', parentType='folder',
-                                          public=False)
+            module_name, func_name = summary_func.rsplit('.',1)
+            module = importlib.import_module(module_name)
+            summary_func = getattr(module, func_name)
+        except:
```

Comment:
`except (AttributeError, ModuleNotFoundError):` ?

Reply:
Not sure it matters? We re-raising the exception anyway. Basically, if anything raises an exception in this block, we want the report to the user that the function could not be loaded and re-raise the exception.